### PR TITLE
fix restart querycoord blocking issue

### PR DESCRIPTION
**Pull request by jiaoew1991**
Signed-off-by: Enwei Jiao <enwei.jiao@zilliz.com>
issue: #18706

### CODE REVIEW

**Comment by jiaoew1991** [comment]
rerun ut

**Comment by jiaoew1991** [comment]
rerun ut

**Comment by jiaoew1991** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

File: internal/querycoord/task_scheduler.go [hunk 1]
```diff
@@ -272,23 +282,27 @@ func (scheduler *TaskScheduler) reloadFromKV() error {
...
-		scheduler.triggerTaskQueue.addTask(t)
-	}
 
-	if doneTriggerTask != nil {
-		scheduler.triggerTaskQueue.addTaskToFront(doneTriggerTask)
-	}
+		if doneTriggerTask != nil {
+			scheduler.triggerTaskQueue.addTaskToFront(doneTriggerTask)
+		}
+	}()
```
Comment (congqixia) [inline internal/querycoord/task_scheduler.go:305]: Is it ok to have the recovery logic fly wild and let querycoord become healthy?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**